boards: arm: nrf51_pca10028: rename board to nrf51dk_nrf51422

ioannisg opened the pull request:
We rename the nRF51 Dev Kit board target (nrf51_pca10028)
to nrf51dk_nrf51422. We update all associated references
in the supportive documentation.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
boards: arm: nrf51_pca10028: rename board to nrf51dk_nrf51422

We rename the nRF51 Dev Kit board target (nrf51_pca10028)
to nrf51dk_nrf51422. We update all associated references
in the supportive documentation and all nRF51-related
cofigurations and overlay files in the samples and tests
in the tree.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

anangl commented on boards/arm/nrf51dk_nrf51422/Kconfig.defconfig:
```diff
@@ -1,12 +1,12 @@
 
 config BOARD
-	default "nrf51_pca10028"
+	default "nrf51dk_nrf51422"
```
You need to also rename all `nrf51_pca10028.overlay` files.

ioannisg replied:
Yes, thanks. And not only this one, apparently.

anangl reviewed (approved):
LGTM

pabigot commented on boards/arm/nrf51dk_nrf51422/doc/index.rst:
```diff
@@ -1,13 +1,13 @@
-The nRF51 Development Kit (PCA10028) hardware provides support for the Nordic
-Semiconductor nRF51822 ARM Cortex-M0 CPU and the following devices:
+The nRF51 Development Kit (nRF51 DK) hardware provides support for the Nordic
+Semiconductor nRF51422 ARM Cortex-M0 CPU and the following devices:
```
It would be nice if these renames retained the association of the old identifier in at minimum the documentation, and possibly other places.  Something as simple as "In earlier Zephyr releases this board was known as `nrf51_pca10028`".  Some people will only know these boards by this name, not the marketing DK name.

This is useful even if we have perfect toolchain infrastructure to allow continued use of the existing name.

ioannisg replied:
Yes, I think we could add a sentence here as you propose.

ioannisg replied:
Will add a separate commit for that, to be reviewed.

ioannisg replied:
Added a note @pabigot .

tejlmand committed:
cmake: deprecation of board names

This commit introduces boards/deprecated.cmake to allow deprecation
of existing boards, when a board is renamed.

This allows users to still specify the old board name, and let Zephyr
build system to select the new board name.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand reviewed (approved):
LGTM.

Did searches without finding missing renames.

SebastianBoe commented on boards/deprecated.cmake:
```diff
@@ -0,0 +1,11 @@
+# time prints a warning to the user, that the board name is deprecated.
+#
+# To add a board rename, add a line in following format:
+# set(<old_board_name>_DEPRECTAED <new_board_name>)
```
typo, DEPRECTAED

tejlmand replied:
fixed

SebastianBoe commented on cmake/app/boilerplate.cmake:
```diff
@@ -181,6 +181,13 @@ endif()
+if(${BOARD}_DEPRECATED)
+  set(BOARD_DEPRECATED ${BOARD})
+  set(BOARD ${${BOARD}_DEPRECATED})
+  message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
```
Let's move the message one line up so we don't have to mutate BOARD_DEPRECATED.

SebastianBoe commented on cmake/app/boilerplate.cmake:
```diff
@@ -181,6 +181,13 @@ endif()
+if(${BOARD}_DEPRECATED)
+  set(BOARD_DEPRECATED ${BOARD})
+  set(BOARD ${${BOARD}_DEPRECATED})
+  message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
```
Let's move the message one line up so we don't have to mutate BOARD_DEPRECATED.

tejlmand replied:
Now that we are using `BOARD_DEPRECATED`, then I don't think we gain anything by moving those lines around.
https://github.com/zephyrproject-rtos/zephyr/blob/faa9b1fdd727f0c32d96a7991728c5fda1348c47/cmake/app/boilerplate.cmake#L161

tejlmand replied:
Just a small informative update.
The original reason I added `BOARD_DEPRECATED` was because I expected that other parts of Zephyr or downstream forks could find it useful to know if a deprecated board was specified and what the value were.

SebastianBoe commented on cmake/app/boilerplate.cmake:
```diff
@@ -181,6 +181,13 @@ endif()
+  set(BOARD ${${BOARD}_DEPRECATED})
+  message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
+endif()
+
```
This now produces an unwanted warning about pristinely cleaning the build dir when we do this:

```
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ cmake -H. -Bb -DBOARD=nrf51_pca10028
-- Application: /home/sebo/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.2.99
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Board: nrf51_pca10028
CMake Warning at /home/sebo/ncs/zephyr/cmake/app/boilerplate.cmake:188 (message):
  Deprecated BOARD=nrf51_pca10028 name specified, board automatically changed
  to: nrf51dk_nrf51422.
Call Stack (most recent call first):
  CMakeLists.txt:5 (include)


-- Found west: /home/sebo/.local/bin/west (found suitable version "0.6.3", minimum required is "0.6.0")
-- Found toolchain: zephyr (/home/sebo/zephyr-sdk-0.11.2)
-- Loading /home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422.dts as base
Devicetree header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/devicetree_unfixed.h'
Parsing /home/sebo/ncs/zephyr/Kconfig
Loaded configuration '/home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422_defconfig'
Merged configuration '/home/sebo/ncs/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
Kconfig header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/sebo/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Cache files will be written to: /home/sebo/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sebo/ncs/zephyr/samples/hello_world/b
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ touch CMakeLists.txt 
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ ninja -C b/
ninja: Entering directory `b/'
[0/1] Re-running CMake...
-- Application: /home/sebo/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.2.99
CMake Warning at /home/sebo/ncs/zephyr/cmake/app/boilerplate.cmake:162 (message):
  The build directory must be cleaned pristinely when changing boards
Call Stack (most recent call first):
  CMakeLists.txt:5 (include)


-- Board: nrf51dk_nrf51422
-- Found west: /home/sebo/.local/bin/west (found suitable version "0.6.3", minimum required is "0.6.0")
-- Found toolchain: zephyr (/home/sebo/zephyr-sdk-0.11.2)
-- Loading /home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422.dts as base
Devicetree header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/devicetree_unfixed.h'
Parsing /home/sebo/ncs/zephyr/Kconfig
Loaded configuration '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
No change to configuration in '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
No change to Kconfig header in '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/autoconf.h'
-- Cache files will be written to: /home/sebo/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sebo/ncs/zephyr/samples/hello_world/b

```

SebastianBoe replied:
I can't figure out how to fix this, so I think we should just live with it.

tejlmand replied:
interesting, cause the cached board should be the switched one, ie `nrf51dk_nrf51422`.

Will tru to look into a fix, cause this is not nice to users.

Thanks foir spotting.

tejlmand replied:
Fixed.
Please recheck, and see if the behavior is as you like.

SebastianBoe commented:
We need to patch mcumgr and mcuboot:

```
/home/sebo/ncs/modules/lib/mcumgr/samples/smp_svr/zephyr/sample.yaml
10:    platform_whitelist: nrf51_pca10028

```

```
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ find ~/ncs/ -name "*10028*"
/home/sebo/ncs/bootloader/mcuboot/boot/zephyr/boards/nrf51_pca10028.conf

```

ioannisg committed:
boards: arm: nrf51dk_nrf51422: add note about the previous board name

Add a note in the board documentation with the nRF51 DK
board was named in earlier Zephyr releases (nrf51_pca10028).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented on boards/deprecated.cmake:
```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: Apache-2.0
```
@tejlmand can you add a set of CODE-OWNERs for this file?

tejlmand replied:
done

ioannisg commented:
> We need to patch mcumgr and mcuboot:
> 
> ```
> /home/sebo/ncs/modules/lib/mcumgr/samples/smp_svr/zephyr/sample.yaml
> 10:    platform_whitelist: nrf51_pca10028
> ```
> 
> ```
> sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ find ~/ncs/ -name "*10028*"
> /home/sebo/ncs/bootloader/mcuboot/boot/zephyr/boards/nrf51_pca10028.conf
> ```

@nvlsianpu could you do that?

nvlsianpu commented:
@ioannisg sure, I can.

nvlsianpu commented:
https://github.com/JuulLabs-OSS/mcuboot/pull/684

nvlsianpu commented:
``` console
/home/sebo/ncs/modules/lib/mcumgr/samples/smp_svr/zephyr/sample.yaml
10:    platform_whitelist: nrf51_pca10028
```

This is not part of zephyr CI build really.

nvlsianpu commented:
https://github.com/apache/mynewt-mcumgr/pull/73

ioannisg commented:
> ```
> /home/sebo/ncs/modules/lib/mcumgr/samples/smp_svr/zephyr/sample.yaml
> 10:    platform_whitelist: nrf51_pca10028
> ```
> 
> This is not part of zephyr CI build really.

But doesn't it need to be fixed @nvlsianpu eventually?

SebastianBoe commented:
@ioannisg : Fix is here: https://github.com/apache/mynewt-mcumgr/pull/73/files

ioannisg committed:
west: update MCUboot SHA

Update MCUboot SHA to take into account the
update of the nRF51 board target name.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

carlescufi commented:
@nashif @MaureenHelm @galak any objections to this? Sets a precedent for board renaming, includes a 'deprecated' board name list that can be used to warn users.

carlescufi commented:
> > ```
> > /home/sebo/ncs/modules/lib/mcumgr/samples/smp_svr/zephyr/sample.yaml
> > 10:    platform_whitelist: nrf51_pca10028
> > ```
> > 
> > 
> > This is not part of zephyr CI build really.
> 
> But doesn't it need to be fixed @nvlsianpu eventually?

@nvlsianpu ^^

nvlsianpu commented:
@carlescufi @ioannisg Eventually yes, that why i opened fix PR to the mcumgr upsterem. Although no need to wait on it here.

ioannisg commented:
> @carlescufi @ioannisg Eventually yes, that why i opened fix PR to the mcumgr upsterem. Although no need to wait on it here.

@nvlsianpu this PR pulls mcuboot from an upstream PR? Would you want to merge that first?

nashif commented on doc/conf.py:
```diff
@@ -337,7 +337,7 @@
 		('boards/arm/nrf51_ble400/doc/nrf51_ble400', 'boards/arm/nrf51_ble400/doc/index'),
 		('boards/arm/nrf51_blenano/doc/nrf51_blenano', 'boards/arm/nrf51_blenano/doc/index'),
-		('boards/arm/nrf51_pca10028/doc/nrf51_pca10028', 'boards/arm/nrf51_pca10028/doc/index'),
+		('boards/arm/nrf51dk_nrf51422/doc/nrf51dk_nrf51422', 'boards/arm/nrf51dk_nrf51422/doc/index'),
```
this looks wrong, it will not create a redirect from old page to the new one.

nashif replied:
You need two entries:

```
('boards/arm/nrf51_pca10028/doc/nrf51_pca10028', 'boards/arm/nrf51dk_nrf51422/doc/index'),
('boards/arm/nrf51_pca10028/doc/nrf51_pca10028/index', 'boards/arm/nrf51dk_nrf51422/doc/index')

```

I think...

ioannisg replied:
updated, thanks
